src/Ydb.Sdk/src/Services/Query/Tx.cs

```diff
@@ -0,0 +1,167 @@
+using Ydb.Query;
+using Ydb.Sdk.Value;
+
+namespace Ydb.Sdk.Services.Query;
+
+public interface ITxModeSettings
+{
+}
+
+public class TxModeSerializableSettings : ITxModeSettings
+{
+}
+
+public class TxModeOnlineSettings : ITxModeSettings
+{
+    public TxModeOnlineSettings(bool allowInconsistentReads = false)
+    {
+        AllowInconsistentReads = allowInconsistentReads;
+    }
+
+    public bool AllowInconsistentReads { get; }
+}
+
+public class TxModeStaleSettings : ITxModeSettings
+{
+}
+
+public class TxModeSnapshotSettings : ITxModeSettings
+{
+}
+
+public class Tx
+{
+    internal QueryClient? QueryClient { private get; set; }
+    internal string? SessionId { private get; set; }
+
+    public string? TxId => _proto.TxId;
+
+    private readonly TransactionControl _proto;
+
+    public Tx()
+    {
+        _proto = new TransactionControl();
+    }
+
+    internal Tx(TransactionControl proto)
+    {
+        _proto = proto;
+    }
+
+    internal TransactionControl ToProto()
+    {
+        return _proto.Clone();
+    }
+
+    public static Tx Begin(ITxModeSettings? txModeSettings = null, bool commit = true)
+    {
+        txModeSettings ??= new TxModeSerializableSettings();
+
+        var txSettings = GetTransactionSettings(txModeSettings);
+
+        var tx = new Tx(new TransactionControl { BeginTx = txSettings, CommitTx = commit });
+        return tx;
+    }
+
+    private static TransactionSettings GetTransactionSettings(ITxModeSettings txModeSettings)
+    {
+        var txSettings = txModeSettings switch
+        {
+            TxModeSerializableSettings => new TransactionSettings
+            {
+                SerializableReadWrite = new SerializableModeSettings()
+            },
+            TxModeOnlineSettings onlineModeSettings => new TransactionSettings
+            {
+                OnlineReadOnly = new OnlineModeSettings
+                {
+                    AllowInconsistentReads = onlineModeSettings.AllowInconsistentReads
+                }
+            },
+            TxModeStaleSettings => new TransactionSettings
+            {
+                StaleReadOnly = new StaleModeSettings()
+            },
+            TxModeSnapshotSettings => new TransactionSettings
+            {
+                SnapshotReadOnly = new SnapshotModeSettings()
+            },
+            _ => throw new InvalidCastException(nameof(txModeSettings))
+        };
+        return txSettings;
+    }
+
+    public async Task<QueryResponseWithResult<T>> Query<T>(
+        string queryString,
+        Dictionary<string, YdbValue>? parameters,
+        Func<ExecuteQueryStream, Task<T>> func,
+        ExecuteQuerySettings? executeQuerySettings = null)
+    {
+        if (QueryClient is null)
```

Comment:
Why do you need check exists of QueryClient and SessionID at runtime instead of require them for constructor?